kratos/includes/registry_item.h

```diff
@@ -66,6 +67,131 @@ class RegistryItem
     /// Pointer definition of RegistryItem
     KRATOS_CLASS_POINTER_DEFINITION(RegistryItem);
 
+    /// Subregistry item type definition
+    using SubRegistryItemType = std::unordered_map<std::string, Kratos::shared_ptr<RegistryItem>>;
+
+    /// Custom iterator with key as return type to be used in the Python export
+    class key_return_iterator
+    {
+    public:
+        ///@name Type Definitions
+        ///@{
+
+        using BaseIterator      = SubRegistryItemType::iterator;
+        using iterator_category = BaseIterator::iterator_category;
```

Comment:
```std::map```'s iterator is ~~random access~~ **bi-directional**, but ```key_return_iterator``` only implements increment operators.
```suggestion
        using iterator_category = std::forward_iterator_tag;
```

Reply:
Agree and done. Thanks for the comment.